cylc/rose/rose.py

```diff
@@ -48,34 +57,108 @@ def get_rose_vars(dir_=None, opts=None):
     config = {
         'env': None,
-        'empy:suite.rc': None,
-        'jinja2:suite.rc': None
+        'template variables': None,
+        'templating detected': None
     }
-    # Return None if dir_ does not exist
+    # Return a blank config dict if dir_ does not exist
     if not rose_config_exists(dir_):
         return config
 
-    # Load the config tree
+    # Load the raw config tree
     config_tree = rose_config_tree_loader(dir_, opts)
 
-    # For each of the template language sections...
-    for section in ['jinja2:suite.rc', 'empy:suite.rc', 'env']:
-        if section in config_tree.node.value:
-            config[section] = dict(
-                [
-                    (item[0][1], item[1].value) for
-                    item in config_tree.node.value[section].walk()
-                ]
-            )
-
+    templating = None
+    if (
+        'jinja2:suite.rc' in config_tree.node.value and
+        'empy:suite.rc' in config_tree.node.value
+    ):
+        raise MultipleTemplatingEnginesError(
+            "You should not define both jinja2 and empy in the same "
+            "configuration file."
+        )
+    elif 'jinja2:suite.rc' in config_tree.node.value:
+        templating = 'jinja2:suite.rc'
+    elif 'empy:suite.rc' in config_tree.node.value:
+        templating = 'empy:suite.rc'
+    if templating:
+        config['templating detected'] = templating
+
+    # Get Values for standard ROSE variables.
+    rose_orig_host = get_host()
+    rose_site = ResourceLocator().get_conf().get_value(['site'], '')
+
+    # Create env section if it doesn't already exist.
+    if 'env' not in config_tree.node.value:
+        config_tree.node.set(['env'])
+
+    # For each section add standard variables and process variables.
+    for section in ['env', templating]:
+        if section not in config_tree.node.value:
+            continue
+
+        # Add standard ROSE_VARIABLES
+        config_tree.node[section].set(['ROSE_SITE'], rose_site)
+        config_tree.node[section].set(['ROSE_VERSION'], ROSE_VERSION)
+        config_tree.node[section].set(['ROSE_ORIG_HOST'], rose_orig_host)
+
+        # Use env_var_process to process variables which may need expanding.
+        for key, node in config_tree.node.value[section].value.items():
+            try:
+                config_tree.node.value[
+                    section
+                ].value[key].value = env_var_process(node.value)
+                if section == 'env':
+                    os.environ[key] = node.value
+            except UnboundEnvironmentVariableError as exc:
+                raise ConfigProcessError(['env', key], node.value, exc)
+
+    # For each of the template language sections extract items to a simple
+    # dict to be returned.
+    if 'env' in config_tree.node.value:
+        config['env'] = dict(
+            [
+                (item[0][1], item[1].value) for
+                item in config_tree.node.value['env'].walk()
+            ]
+        )
```

Comment:
:snake: :snake: :snake: 

```suggestion
        config['env'] = {
            item[0][1]: item[1].value)
            for item in config_tree.node.value['env'].walk()
        }
```